packages/contractkit/src/wrappers/LockedGold.ts

```diff
@@ -50,6 +50,7 @@ export class LockedGoldWrapper extends BaseWrapper<LockedGold> {
   maxNoticePeriod = proxyCall(this.contract.methods.maxNoticePeriod, undefined, toBigNumber)
 
   getAccountWeight = proxyCall(this.contract.methods.getAccountWeight, undefined, toBigNumber)
+  getDelegateFromAccountAndRole = proxyCall(this.contract.methods.getDelegateFromAccountAndRole)
```

Comment:
see above. make it expect a `Role`, not a index.

Also, document

Reply:
What's the way to enforce parameter types in `proxyCall`?

Reply:
Check the docs for `proxyCall`

```ts
/**
 * Creates a proxy to call a web3 native contract method.
 *
 * There are 4 cases:
 *  - methodFn
 *  - parseInputArgs => methodFn
 *  - parseInputArgs => methodFn => parseOutput
 *  - methodFn => parseOutput
 *
 * @param methodFn Web3 methods function
 * @param parseInputArgs [optional] parseInputArgs function, tranforms arguments into `methodFn` expected inputs
 * @param parseOutput [optional] parseOutput function, transforms `methodFn` output into proxy return
 */
```

you can define `parseInputArgs` and transform a Role to an index for the actual contract call.

You can check on other wrappers for examples

Reply:
Just saw your Role enum def, you can do:

```ts
getDelegateFromAccountAndRole: (...args here) => Promise<ResultHere>
```

See for example:
```ts
  /**
   * @dev Returns the amount of buyToken a user would get for sellAmount of sellToken
   * @param sellAmount The amount of sellToken the user is selling to the exchange
   * @param sellGold `true` if gold is the sell token
   * @return The corresponding buyToken amount.
   */
  getBuyTokenAmount: (sellAmount: NumberLike, sellGold: boolean) => Promise<BigNumber> = proxyCall(
    this.contract.methods.getBuyTokenAmount,
    tupleParser(parseNumber, identity),
    toBigNumber
  )
```